### DISCUSSION TITLE
Need to create a new opportunity in NS ??

### DISCUSSION THREAD

**Discussion by darren-kimberbell** [netsuitephp/netsuite-php#120] [Q&A] [answered]
Hello All,  

I hope that I am asking this in the right place.  I can see how to search opportunities using the PHP toolkit, and that works well, but for some reason I am having trouble trying to figure out how to use this library to create a new opportunity.  

If someone could help to point me in the right direction then I would really appreciate it. 

Thanks so much!
Darrren Nay
darren@kimberbell.com

**Comment by jrebs** [comment]
I've never even heard of opportunities before, but it looks like the `Opportunity` class just extends the `Record` class like others such as `CashSale` or `Address`, so I'm unclear on why creating a new opportunity would be any different than creating any other record. Is the issue that you don't know how to create records in the first place or is there something uniquely wrong with creating opportunities? If the latter, please provide more info and maybe we can glean something from it. If the former, see [EXAMPLES](https://github.com/netsuitephp/netsuite-php/blob/master/EXAMPLES.md) for a few samples of adding a record and in theory this shouldn't be fundamentally different. Web services responses will generally give you fairly easy-to-understand messages about fields you need to provide (or mustn't provide), etc.

**Comment by darren-kimberbell** [comment, marked as answer]
Thanks Justin,  I appreciate your response!  You are correct.  It is the same.  I have gotten it to work using something similar to the following code snippet.

 	// Create the NS Opportunity object
        $opportunity = new Opportunity();

        $opportunity->subsidiary = array('internalId' => 1); 
        $opportunity->class = array('internalId' => 46); 
        $opportunity->entity = array('internalId' => 16); 
        $opportunity->currencyName = "USA";
        $opportunity->exchangeRate = 1.00;

        // Build the create request and send it to NetSuite API
        $request = new AddRequest();
        $request->record = $opportunity;
        $response = $service->add($request);
